### ISSUE TITLE
Reordering components inside a component in the Content Manager yields an error

### ISSUE THREAD

**Issue by grdevd** [strapi/strapi#15764]
### Required System information

- Node.js version: 16.14.2
- NPM version: 8.5.5
- Strapi version: 4.6.1
- Database: PostgreSQL
- Operating system: MacOS/Linux
- Is your project Javascript or Typescript: Js

### Describe the bug

Reordering components inside a component in the Content Manager yields an error. See the attached video.

### Steps to reproduce the behavior

1. Create a structure where a collection has a list of components within a list of components.
2. In the content manager, try reordering the components, that are inside a component of the item of the collection.

### Expected behavior

Reordering items should work without raising an error.

### Screenshots / Screen recording

[Screen record](https://user-images.githubusercontent.com/94918262/217844816-6f3a748d-c7e8-4856-a7e0-4abce151ed62.mp4)

### Code snippets

/

### Additional context / stack trace

```
main.7e97da24.js:875 TypeError: Cannot read properties of undefined (reading '__temp_key__')
    at content-manager.e89578a7.chunk.js:65:2711
    at Array.map (<anonymous>)
    at xt (content-manager.e89578a7.chunk.js:65:2702)
    at content-manager.e89578a7.chunk.js:829:408
    at Object._l [as useMemo] (main.7e97da24.js:873:59627)
    at n.useMemo (main.7e97da24.js:894:5833)
    at An (content-manager.e89578a7.chunk.js:829:403)
    at Zl (main.7e97da24.js:873:54675)
    at vu (main.7e97da24.js:873:64509)
    at Hc (main.7e97da24.js:877:10061)
Eu @ main.7e97da24.js:875
N.payload @ main.7e97da24.js:875
Ir @ main.7e97da24.js:873
Zu @ main.7e97da24.js:873
Hc @ main.7e97da24.js:877
Bc @ main.7e97da24.js:877
qc @ main.7e97da24.js:877
cu @ main.7e97da24.js:877
Iu @ main.7e97da24.js:875
(anonymous) @ main.7e97da24.js:873
n.unstable_runWithPriority @ main.7e97da24.js:901
eo @ main.7e97da24.js:873
Gs @ main.7e97da24.js:873
Oa @ main.7e97da24.js:873
is @ main.7e97da24.js:875
ql @ main.7e97da24.js:873
(anonymous) @ content-manager.e89578a7.chunk.js:960
Z @ content-manager.e89578a7.chunk.js:829
hover @ content-manager.e89578a7.chunk.js:65
hover @ main.7e97da24.js:862
(anonymous) @ main.7e97da24.js:467
l @ main.7e97da24.js:467
(anonymous) @ main.7e97da24.js:467
(anonymous) @ main.7e97da24.js:467
handleTopDragEnter @ 6914.897b0063.chunk.js:1
```

```
main.7e97da24.js:467 Uncaught TypeError: Cannot read properties of undefined (reading 'hover')
    at main.7e97da24.js:467:36254
    at Array.forEach (<anonymous>)
    at l (main.7e97da24.js:467:36220)
    at a.<anonymous> (main.7e97da24.js:467:35581)
    at Object.hover (main.7e97da24.js:467:38954)
    at handleTopDragEnter (6914.897b0063.chunk.js:1:13578)
(anonymous) @ main.7e97da24.js:467
l @ main.7e97da24.js:467
(anonymous) @ main.7e97da24.js:467
(anonymous) @ main.7e97da24.js:467
handleTopDragEnter @ 6914.897b0063.chunk.js:1
```

**Comment by Imran-cse** [comment]
Facing same issue

**Comment by grdevd** [comment]
@derrickmehaffy  Can I help giving some more context or maybe examples to get to the root of the problem? Currently, editing items in the Content Manager is limited due to this bug and this is slowly becoming a problem.

**Comment by sleeporsmile** [comment]
+1

**Comment by sleeporsmile** [comment]
> @derrickmehaffy Can I help giving some more context or maybe examples to get to the root of the problem? Currently, editing items in the Content Manager is limited due to this bug and this is slowly becoming a problem.
> 
> It's still occuring with the latest Strapi version 4.9.0

[16328](https://github.com/strapi/strapi/issues/16328)

**Comment by sleeporsmile** [comment]
https://github.com/strapi/strapi/issues/16328

**Comment by Harasz** [comment]
I've created a patch that fix the problem with `TypeError`, however it produce other problems like: reordering parent component and creating empty components on parent level.

```
diff --git a/node_modules/@strapi/admin/admin/src/content-manager/components/RepeatableComponent/index.js b/node_modules/@strapi/admin/admin/src/content-manager/components/RepeatableComponent/index.js
index ffa517c..6e09e4f 100644
--- a/node_modules/@strapi/admin/admin/src/content-manager/components/RepeatableComponent/index.js
+++ b/node_modules/@strapi/admin/admin/src/content-manager/components/RepeatableComponent/index.js
@@ -197,7 +197,9 @@ const RepeatableComponent = ({
       <VisuallyHidden aria-live="assertive">{liveText}</VisuallyHidden>
       <Accordion.Group error={errorMessage} ariaDescribedBy={ariaDescriptionId}>
         <Accordion.Content aria-describedby={ariaDescriptionId}>
-          {componentValue.map(({ __temp_key__: key }, index) => (
+          {componentValue
+          .map(v => ({ ...(v ?? {}), __temp_key__: v?.__temp_key__ ?? -1 }))
+          .map(({ __temp_key__: key }, index) => (
             <Component
               componentFieldName={`${name}.${index}`}
               componentUid={componentUid}
diff --git a/node_modules/@strapi/admin/admin/src/content-manager/utils/getMaxTempKey.js b/node_modules/@strapi/admin/admin/src/content-manager/utils/getMaxTempKey.js
index 8a73df0..856c97f 100644
--- a/node_modules/@strapi/admin/admin/src/content-manager/utils/getMaxTempKey.js
+++ b/node_modules/@strapi/admin/admin/src/content-manager/utils/getMaxTempKey.js
@@ -5,7 +5,7 @@ const getMaxTempKey = (arr) => {
 
   const maxTempKey = Math.max.apply(
     Math,
-    arr.map((o) => o.__temp_key__ ?? 0)
+    arr.map(o => o?.__temp_key__ ?? 0)
   );
 
   return Number.isNaN(maxTempKey) ? -1 : maxTempKey;
```

**Comment by TheBit** [comment]
Is it the same? Fixed now? https://github.com/strapi/strapi/pull/16332

**Comment by joshuaellis** [comment]
> Is it the same? Fixed now? #16332 / [`v4.9.2` (release)](https://github.com/strapi/strapi/releases/tag/v4.9.2)

No, the fix for this issue is #16441
